### PR TITLE
Tm th delete items

### LINKED ISSUE #12: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
It’s possible our users may make an error when entering an item or may decide they won’t be buying a certain item again in the future. Let’s help them keep their lists tidy!

AC:

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

### PR DESCRIPTION
**Pull request by tarynblakemiller**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description

This branch allows a user to delete an item from their list by click the `delete` button adjacent to each item. When a user clicks the button they are first prompted to confirm whether or not they meant to delete this item.

## Related Issue

Closes #12 

## Acceptance Criteria

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal window pull down the related branch using the command `git pull origin tm-th-delete-items` and then migrate to that branch using `git checkout tm-th-delete-items`

* Install dependencies using `npm install`

* Start the server with `npm start`

* Navigate to the list view to view your items

* Delete an item from your list and firestore by clicking the red `Delete` button 

* You should now no longer see the item in the list as well as in the Firestore

### CODE REVIEW

**Comment by ashill99** [comment]
All looks good to me! Only query would be whether we want to keep delete button red as it becomes a bit less clear on a red item background. Maybe we switch the shades of one of the reds?

**Review by ashill99 — APPROVED** [review]

File: src/components/GroceryItem/GroceryItem.js [hunk 1]
```diff
@@ -94,6 +105,19 @@ const GroceryItem = ({ item }) => {
...
         />
         {item.name} - Repurchase in {item.howSoon} days
+        <button
+          className="delete-button"
+          onClick={() => {
+            const confirmation = window.confirm(
+              'Did you mean to delete this item?',
+            );
+            if (confirmation) {
+              deleteItem(item.id);
```
Comment (vanmars) [inline src/components/GroceryItem/GroceryItem.js]: For readability, should we move this function up above the return. Then you can just change line 110 to: onClick={confirmation}

**Review by vanmars — APPROVED** [review]
Looks great! Nice use of a confirm box. Just one non-blocking comment for readability.

**Review by jandrade — APPROVED** [review]
Looks good to me! +1 to what Vanessa and Adam said in the PR, thanks for the clear implementation :)

**Pull request merged**